### PR TITLE
small fixes and add support for noblock pushing

### PR DESCRIPTION
**Pull request by mediocregopher**
I also changed the callback function to only take in a single argument which is an assoc-array with all the info we want to give the consumer on it. Before it was a queue name and an array which had the id and the contents; with this way there's less to think about and we can easily add more stuff in the future without breaking anything.

### CODE REVIEW

**Comment by mc0** [comment]
I guess I didn't specify but I was trying to keep PHP 5.2+ compat

**Comment by mediocregopher** [comment]
fixed

File: okq/Okq.php [hunk 1]
```diff
@@ -134,32 +132,27 @@ public function consume($callback, $queues = null, $timeout = 30)
...
-                    }
-                    break;
-                case self::EVENT_STOP:
-                    $continue = false;
-                    break;
-                case self::EVENT_NOACK:
-                default:
-                    break;
+            $response = call_user_func($callback, $event);
+            if (($response & self::EVENT_ACK) == self::EVENT_ACK) {
```
Comment (mc0) [inline okq/Okq.php]: needs strict equality

File: okq/Okq.php [hunk 2]
```diff
@@ -134,32 +132,27 @@ public function consume($callback, $queues = null, $timeout = 30)
...
+                if (isset($event['id'])) {
+                    $eventId = (string)$event['id'];
+                    $this->qack($queue, $eventId);
+                }
+            }
+            if (($response & self::EVENT_STOP) == self::EVENT_STOP) {
```
Comment (mc0) [inline okq/Okq.php]: strict eq

**Comment by mediocregopher** [comment]
fixed again

**Pull request merged**